### PR TITLE
fix(docbuild): reformat comments for Doxygen in `lv_conf_template.h`

### PR DESCRIPTION
**Pull request by vwheeler63**
### Description of the feature or fix

There are several files closely connected with `lv_conf_template.h` and they are coordinated and changed together with this PR.
- Reformat comments of `lv_conf_template.h` for Doxygen.
- Add 2 indents in nested `#if`s to be consistent with rest of file.
- Add a few Doxygen comments that will help in API documentation.
- Make several clarifications to make using of `lv_conf.h` more friendly.
- Update `lv_conf_internal_gen.py` with 1 Doxygen comment and add spacing in comments it adds for consistency with `lv_conf_template.h`.
- Run `lv_conf_internal_gen.py` so it carries above changes to `lv_conf_internal.h`.
- Note: Kconfig is unaffected by this PR.

Addresses part of issue #6480.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by vwheeler63** [comment]
@kisvegabor 

Ready for review.

Note that it is probably easier to look at the WHOLE FILE (`lv_conf_template.h`) rather than just the changes, since they are extensive.  Most of the comments therein were affected, and I added indentations in 2 places in code nested in `#if MACRO_NAME/#endif` to be consistent with the rest of the file.  And I added new Doxygen-format comments for the `LV_LOG_TRACE_...` macros to explain briefly what they do both for programmers looking at `lv_conf.h`, and also for the generated API documentation.

The changes to `lv_conf_internal_gen.py` ONLY affect the comments it inserts into the generated `lv_conf_internal.h`.

And __lv_conf.h__ is an exact copy of `lv_conf_template.h` save `#if 0` is `#if 1` at the top of the file.

A note on the `LV_USE_STDLIB_MALLOC`, `LV_USE_STDLIB_STRING`, and `LV_USE_STDLIB_SPRINTF` macros at the top, I realize my copy/pasting the original "group" comment is redundant, but it adds that information to each of those (important) macros in the generated API documentation, so I felt it was a worthwhile sacrifice (at the expense of redundancy in the `lv_conf_template.h` file.  That's the only place I did that (added redundancy for a documentation trade-off).

As for the clarifications (wording changes), I think you will find them all accurate.

File: lv_conf_template.h [hunk 1]
```diff
@@ -581,380 +625,404 @@
     #define LV_USE_CALENDAR_CHINESE 0
 #endif  /*LV_USE_CALENDAR*/
 
+/** Include ./src/widgets/lv_canvas* logic in compile. */
```
Comment (kisvegabor) [inline lv_conf_template.h]: Why are these needed?

File: lv_conf_template.h [hunk 2]
```diff
@@ -581,380 +625,404 @@
...
-/*1: Enable Monkey test*/
+/** 1: Enable Monkey test */
 #define LV_USE_MONKEY 0
 
-/*1: Enable grid navigation*/
+/** 1: Enable grid navigation */
 #define LV_USE_GRIDNAV 0
 
-/*1: Enable lv_obj fragment*/
+/** 1: Enable `lv_obj` fragment logic (@see  https://docs.lvgl.io/master/others/fragment.html ) */
```
Comment (kisvegabor) [inline lv_conf_template.h]: Why did you add a link only here?

**Review by kisvegabor — COMMENTED** [review]
Thank you Vic! 
I have only  2 questions. (See below)

Regarding `__lv_conf.h__` please keep it as it is. I'll take care of it separately.

File: lv_conf_template.h (hunk 1 above)
Comment (vwheeler63) [reply lv_conf_template.h]: Fair question.  My thought at that moment was that it would be a good idea to include specifically what these do so the programmer did not have to scan for the symbols to find out.  However, in retrospect, they _might_ be somewhat self-explanatory.  When I was learning v8.x last Fall, I had to scan the code to find out what they really did -- this was when I was looking for a way to reduce the program-size footprint.  So at that time, I would have found the comments helpful -- it would have allowed me to have confidence to turn them off without having to scan.   Would you like me to remove them?

File: lv_conf_template.h (hunk 2 above)
Comment (vwheeler63) [reply lv_conf_template.h]: Good question.  I hadn't heard of `LV_USE_FRAGMENT` before so I took the time to look up in the documentation, and I thought it would be appropriate to bring helpful knowledge back into `lv_conf.h`.  I thought of it as "enhancement by increments".  Are you thinking like I am -- that it would be a good idea to provide links for all of them?

File: lv_conf_template.h (hunk 1 above)
Comment (vwheeler63) [reply lv_conf_template.h]: > Regarding `__lv_conf.h__` please keep it as it is. I'll take care of it separately.

Okay, I will remove that from the PR now.

Comment (kisvegabor) [reply lv_conf_template.h]: > Would you like me to remove them?

It's a trade off. If we add too many comments the config file will be noisy. With too little info, it will be hard to understand. In this particular case I vote for removing these. It should be obvious that `LV_USE_CANVAS` is used in `lv_canvas.c/h`.

File: lv_conf_template.h (hunk 2 above)
Comment (kisvegabor) [reply lv_conf_template.h]: Now the docs main section titles mostly mimics the structure of `lv_conf.h`
- "WIDGETS"
- "OTHERS"
- "3RD PARTY LIBS"

If you really find it useful, I'd add links below the main sections. E.g
```
/*==================
 * OTHERS
 *==================*/
/* @see  https://docs.lvgl.io/master/others/index.html */
```

File: lv_conf_template.h (hunk 1 above)
Comment (vwheeler63) [reply lv_conf_template.h]: Agreed.

File: lv_conf_template.h (hunk 2 above)
Comment (vwheeler63) [reply lv_conf_template.h]: Hi, Gabor!  ( @kisvegabor )

I did find it useful, and think that idea will be helpful for others as well.  So following your guidance, I added documentation links (following the nice example already under WIDGETS) to the major sections below the WIDGETS section:  THEMES, LAYOUTS, 3RD PARTY LIBRARIES and OTHERS.  I did not find a clear connection between the DEVICES section and the documentation, so I left it alone.

Given that the `@see` is only useful in a Doxygen comment that documents a code member (or file or directory), I replaced it with the clear example already under WIDGETS section heading.  So they look like this:

```
/* Documentation for ______ can be found here: https://docs.lvgl.io/master/others/index.html . */
```

Submitted and ready again for review.

**Comment by vwheeler63** [comment]
@kisvegabor Hi, Gabor!  Conflict resolved.  Ready for re-review/approval/merge.  :-)
--Vic

**Review by kisvegabor — DISMISSED** [review]
Thank you! I've just fixed some new conflicts.

**Comment by vwheeler63** [comment]
> Thank you! I've just fixed some new conflicts.

Thank you for that!

File: lv_conf_template.h [hunk 3]
```diff
@@ -26,22 +26,38 @@
    COLOR SETTINGS
  *====================*/
 
-/*Color depth: 1 (I1), 8 (L8), 16 (RGB565), 24 (RGB888), 32 (XRGB8888)*/
+/** Color depth: 8 (A8), 16 (RGB565), 24 (RGB888), 32 (XRGB8888) */
```
Comment (kisvegabor) [inline lv_conf_template.h]: Keep I1 and L8
```suggestion
/** Color depth: 1 (I1), 8 (L8), 16 (RGB565), 24 (RGB888), 32 (XRGB8888) */
```

File: lv_conf_template.h [hunk 4]
```diff
@@ -260,79 +272,87 @@
...
+    /** - 1: Print file and line number of the log;
+     *  - 0: Do not print file and line number of the log. */
     #define LV_LOG_USE_FILE_LINE 1
 
+    /* Enable/disable LV_LOG_TRACE in modules that produces a huge number of logs. */
 
-    /*Enable/disable LV_LOG_TRACE in modules that produces a huge number of logs*/
+    /** Enable/disable trace logs in memory operations. */
     #define LV_LOG_TRACE_MEM        1
+    /** Enable/disable trace logs in timer operations. */
```
Comment (kisvegabor) [inline lv_conf_template.h]: Can we make it inline with the define to save some space?

**Review by kisvegabor — COMMENTED** [review]
Thank you! Just 2 minor comments.

File: lv_conf_template.h (hunk 3 above)
Comment (vwheeler63) [reply lv_conf_template.h]: Good catch, @kisvegabor !  I hadn't realized this has changed for v9.2!  This inspired me to also look to find if there were any more recent changes to `lv_conf_template.h` for v9.2 that I might have overlooked, and I found one more -- the comments to the right of the LV_WAYLAND_W... macros.  Also fixed.

File: lv_conf_template.h (hunk 4 above)
Comment (vwheeler63) [reply lv_conf_template.h]: @kisvegabor Great idea!  Done and submitted.

**Review by kisvegabor — DISMISSED** [review]
Looks good to me! Thank you!

File: lv_conf_template.h [hunk 5]
```diff
@@ -645,338 +671,342 @@
...
 #endif
 
-/*API for Arduino Sd. */
+/** API for Arduino Sd. */
 #define LV_USE_FS_ARDUINO_SD 0
 #if LV_USE_FS_ARDUINO_SD
-    #define LV_FS_ARDUINO_SD_LETTER '\0'     /*Set an upper cased letter on which the drive will accessible (e.g. 'A')*/
+    #define LV_FS_ARDUINO_SD_LETTER '\0'          /**< Set an upper cased letter on which the drive will accessible (e.g. 'A') */
+    #define LV_FS_ARDUINO_SD_CS_PIN 0             /**< Set the pin connected to the chip select line of the SD card */
+    #define LV_FS_ARDUINO_SD_FREQUENCY 40000000   /**< Set the frequency used by the chip of the SD CARD */
```
Comment (liamHowatt) [inline lv_conf_template.h]: These were recently removed

**Review by liamHowatt — COMMENTED** [review]
Thanks for the huge effort!

File: lv_conf_template.h (hunk 5 above)
Comment (vwheeler63) [reply lv_conf_template.h]: > These were recently removed

Ouch!  My apologies.  I know what happened:  these came up in a conflict on my side and I mis-interpreted my merge tool, thinking they had been _added_ instead of _removed_.  Won't happen again if I can help it.  :-)

Re-submitted now.

**Review by liamHowatt — APPROVED** [review]

**Comment by vwheeler63** [comment]
> Thanks for the huge effort!

And thank YOU for putting in the time to examine this long list of changes!  :-)

--Vic

**Review by kisvegabor — APPROVED** [review]
Thank you Vic!
